### ISSUE TITLE
where is the gradient descend

### ISSUE THREAD

**Issue by Bohrhh** [42x00/p3i#4]
I have went through all the code and still not found some code like loss.backward(). So how does p3i do the gradient descend?
